language/diem-framework/src/release.rs

```diff
@@ -190,3 +259,111 @@ pub fn build_error_code_map(output_path: &Path) {
     options.setup_logging_for_test();
     move_prover::run_move_prover_errors_to_stderr(options).unwrap();
 }
+
+pub struct ReleaseOptions {
+    pub check_layout_compatibility: bool,
+    pub build_modules: bool,
+    pub module_docs: bool,
+    pub script_docs: bool,
+    pub with_diagram: bool,
+    pub script_abis: bool,
+    pub script_builder: bool,
+    pub errmap: bool,
+    pub time_it: bool,
+}
+
+impl Default for ReleaseOptions {
+    fn default() -> Self {
+        Self {
+            check_layout_compatibility: false,
+            build_modules: true,
+            module_docs: true,
+            script_docs: true,
+            with_diagram: false,
+            script_abis: true,
+            script_builder: true,
+            errmap: true,
+            time_it: false,
+        }
+    }
+}
+
+fn run_step<F, R>(step_msg: Option<&str>, f: F) -> R
+where
+    F: FnOnce() -> R,
+{
+    match step_msg {
+        Some(msg) => move_stdlib::utils::time_it(msg, f),
+        None => f(),
+    }
+}
+
+pub fn create_release(output_path: impl AsRef<Path>, options: &ReleaseOptions) {
```

Comment:
I suppose the `output_path` here is the version name in the previous framework getter? Probably will need to make that clear in the code (say have the same argument name or some comments)

Reply:
I'll add some comments explaining what some of the public APIs/important functions are doing.